build.js

```diff
@@ -35,16 +35,58 @@ async function rmDistDir() {
   }
 }
 
-async function main() {
-  console.log('Compiling sequelize...');
-  const [declarationFiles, filesToCompile] = await Promise.all([
+function sourcePathToDestPath(source) {
+  return path.resolve(outdir, source).replace(/\.ts$/, '.js');
+}
+
+async function ageCheckFiles(paths) {
+  return (await Promise.all(
+    paths.map(source => new Promise(resolve => {
+      const dest = sourcePathToDestPath(source);
+      const sourceStat = fs.statSync(source, { throwIfNoEntry: false });
+      const destStat = fs.statSync(dest, { throwIfNoEntry: false });
+      const ok = sourceStat && (!destStat || destStat.mtimeMs < sourceStat.mtimeMs);
```

Comment:
Do I understand it correctly, that we will only update the files that have changed?

Reply:
Right, I was going to ask the question: how should I document this so people learn to use the feature ? Perhaps add also a new entry in package.json ? like build-fast or some such ?

Reply:
`build-dev` maybe?

Reply:
`build-dev` sounds like a dev build, something that would typically contain debug symbols and so forth.
Something like `build-fast` or `build-newest` would be more appropriate.

Reply:
`build-incremental` like tsc's flag? I'm also ok with the other names

Reply:
ok, incremental it is